RecoVertex/Configuration/python/RecoVertex_phase2_timing_cff.py

```diff
@@ -76,9 +78,37 @@
 from SimTracker.TrackAssociation.trackTimeValueMapProducer_cfi import trackTimeValueMapProducer
 from RecoMTD.TimingIDTools.tofPIDProducer_cfi import tofPIDProducer
 
-tofPID4DnoPID=tofPIDProducer.clone(vtxsSrc='unsortedOfflinePrimaryVertices4DnoPID')
+tofPID4DnoPID=tofPIDProducer.clone(vtxsSrc='unsortedOfflinePrimaryVertices')
 tofPID=tofPIDProducer.clone()
+tofPID3D=tofPIDProducer.clone(vtxsSrc='unsortedOfflinePrimaryVertices')
 
 from Configuration.Eras.Modifier_phase2_timing_layer_cff import phase2_timing_layer
-phase2_timing_layer.toModify(tofPID, vtxsSrc='unsortedOfflinePrimaryVertices4D')
+phase2_timing_layer.toModify(tofPID, vtxsSrc='unsortedOfflinePrimaryVertices4D', vertexReassignment=False)
+phase2_timing_layer.toModify(tofPID3D, vertexReassignment=False)
+phase2_timing_layer.toModify(unsortedOfflinePrimaryVertices, 
+    vertexCollections = cms.VPSet(
+        cms.PSet(
+            algorithm = cms.string('AdaptiveVertexFitter'),
+            chi2cutoff = cms.double(2.5),
```

Comment:
@slava77 @werdmann @fabiocos while addressing your comments we encountered some inconsistencies in the config definitions. We updated the fillDescription of the PrimaryVertexProducer, but the unsortedOfflinePrimaryVertices (https://github.com/cms-sw/cmssw/blob/master/RecoVertex/Configuration/python/RecoVertex_cff.py#L16) use a different python config (https://github.com/cms-sw/cmssw/blob/master/RecoVertex/PrimaryVertexProducer/python/OfflinePrimaryVertices_cfi.py.
The two configs look like duplicates, with a few differences, can you clarify if both are needed? Or should we adapt the code to use the cfi from the fillDescription only?

Reply:
As far as I can see, the ```fillDescription``` was introduced by @werdmann back in 2020 in #29626 , without touching the explicit configuration fragments that were already present. Should not we have one and only one master definition, to be cloned and modified wherever needed?

I also notice that in the master configuration of ```RecoVertex``` the main ```offlineVerticesProducer``` effectively changes meaning in

https://github.com/cms-sw/cmssw/blob/master/RecoVertex/Configuration/python/RecoVertex_cff.py#L17

I guess this is due to historical reasons, it looks like anyway a bit weird.

Reply:
according to the Framework workbook https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Converting_an_Existing_CFI_file

"Note when you use a CFI file in a configuration passed to cmsRun, a manually written CFI file in the python subdirectory will override a generated CFI file with the same file name, so you probably want to remove the existing CFI file from the CMSSW repository when you create a fillDescriptions function that generates a CFI file with the same name."

So having both is not forbidden, and the code indeed is working since more than 3 years. This also explains why @kdeleo had to manually update also the CFI in the python subdirectory to get it working. But IMHO having both risks to create confusion.

Reply:
looking into the repository, most of the configurations use the python definition ```OfflinePrimaryVertices_cfi```, and it is not difficult to reproduce it as a ```fillDescription```. But there are a few configurations, among them HI and validation, that are directly using the fragment ```primaryVertexProducer_cfi``` produced by the ```fillDescription```. This creates the risk of inconsistencies. The same issue applies to the definitions in ```TkClusParameters_cff```, that are replicating PSets filled independently in different helper classes, and this fragment is independently called in several parts of CMSSW.

Reply:
@slava77 in the last two commits [7a7f05f](https://github.com/cms-sw/cmssw/pull/43592/commits/7a7f05fbc86adb0beed566560d813a69234f3c23) and [4bdf453](https://github.com/cms-sw/cmssw/pull/43592/commits/4bdf453e854367db70043812f247f6c73308638b) Fabio and I have updated the RecoVertex configurations. Now the duplicates are removed and the master fillDescription is used. The parameters have been updated accordingly to be consistent with the python config previously used.